src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
+    private final SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder periodData = new StringBuilder();
+        try {
+            Date fromDate = simpleDateFormat.parse(dateFrom);
+            Date toDate = simpleDateFormat.parse(dateTo);
+            for (String day : data) {
+                String[] string = day.split(" ");
+                Date date = simpleDateFormat.parse(string[0]);
+                if ((date.after(fromDate) || fromDate.equals(date))
+                        && (date.before(toDate) || date.equals(toDate))) {
+                    periodData.append(day).append(",");
+                }
+            }
+        } catch (ParseException e) {
+            System.out.println("Can't parse date");
+        }
```

Comment:
you don't need this logic
just create LocaDate for `dateFrom` and `dateTo` and add check period where you check person name